cardinal/statsd/statsd.go

```diff
@@ -0,0 +1,47 @@
+// Package statsd is a helper package that wraps some common statsd methods.
+// It hides the datadog dependency so if we decide to migrate away from datadog in the future, we only need to
+// edit this single file. For example, the https://pkg.go.dev/github.com/cactus/go-statsd-client/statsd package roughly
+// implements datadog's ClientInterface interface.
+package statsd
+
+import (
+	"time"
+
+	ddstatsd "github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/rs/zerolog/log"
+)
+
+var client ddstatsd.ClientInterface = &ddstatsd.NoOpClient{}
+
+func Client() ddstatsd.ClientInterface {
+	return client
+}
+
+func EmitTickStat(start time.Time, stage string) {
+	duration := time.Since(start)
+	err := Client().Timing("tick", duration, []string{stage}, 1)
+	if err != nil {
+		log.Logger.Warn().Msgf("failed to emit tick stat: %v", err)
+	}
+}
+
+func Init(address string, tags []string) error {
+	if address == "" {
+		return nil
+	}
```

Comment:
this seems worthy of an error

Reply:
My thinking was that stats tracking can be disabled by setting this to an empty string.

Reply:
i think it might be nice to pull that behavior out - check if the address is non-empty before calling `Init`. if it is empty, we could log a warning as well or something?

Reply:
kk, it's an error to try to init statsd with an empty address, and we'll emit a warning when stats are disabled.